### PR TITLE
Parse when condition in blueprint

### PR DESCRIPTION
**Pull request by sbarker2**
We need a code representation of the when statement.

### CODE REVIEW

File: lib/src/main/java/graphql/nadel/engine/blueprint/hydration/NadelHydrationWhenCondition.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package graphql.nadel.engine.blueprint.hydration
+
+import graphql.nadel.engine.transform.query.NadelQueryPath
+
+sealed class NadelHydrationWhenCondition {
+    abstract fun evaluate(resultId: String): Boolean
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/engine/blueprint/hydration/NadelHydrationWhenCondition.kt:6]: `resultId` should be `Any` (maybe nullable too) probably. But we can fix that up in another PR.

File: lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt [hunk 2]
```diff
@@ -265,9 +265,39 @@ private class Factory(
...
                 }
             },
+            condition = getHydrationCondition(hydration),
         )
     }
 
+    private fun getHydrationCondition(): NadelHydrationWhenCondition? {
+        return null
+    }
+
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt:275]:
```suggestion
```

File: lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt [hunk 3]
```diff
@@ -489,7 +520,7 @@ private class Factory(
                 }
                 StaticArgument -> {
                     NadelHydrationActorInputDef.ValueSource.StaticValue(
-                        value =  remoteArgDef.remoteArgumentSource.staticValue!!
+                        value = remoteArgDef.remoteArgumentSource.staticValue!!
```
Comment (sbarker2) [inline lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt:523]: this just from reformatting

File: lib/src/main/java/graphql/nadel/engine/blueprint/hydration/NadelHydrationWhenCondition.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        val regex: Regex,
+    ) : NadelHydrationWhenCondition() {
+        override fun evaluate(resultId: String): Boolean {
+            TODO("Not yet implemented")
+        }
+    }
+
+    data class StringResultStartsWith(
+        val fieldPath: NadelQueryPath,
+        val value: String,
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/engine/blueprint/hydration/NadelHydrationWhenCondition.kt:28]:
```suggestion
        val prefix: String,
```

File: lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt [hunk 5]
```diff
@@ -265,9 +265,39 @@ private class Factory(
...
+        if (hydration.conditionalHydration.predicate.equals != null) {
+            return NadelHydrationWhenCondition.ResultEquals(
+                fieldPath = NadelQueryPath(hydration.conditionalHydration.sourceField),
+                value = hydration.conditionalHydration.predicate.equals
+            )
+        }
+        if (hydration.conditionalHydration.predicate.startsWith != null) {
+            return NadelHydrationWhenCondition.StringResultStartsWith(
+                fieldPath = NadelQueryPath(hydration.conditionalHydration.sourceField),
+                value = hydration.conditionalHydration.predicate.startsWith
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt:289]:
```suggestion
                prefix = hydration.conditionalHydration.predicate.startsWith
```

**Review by gnawf — CHANGES_REQUESTED** [review]
Some pending changes, but looks good otherwise.

**Review by gnawf — APPROVED** [review]

**Pull request merged**